lib/util/env/src/lib.rs

```diff
@@ -5,6 +5,20 @@ pub async fn read_secret(key: &[impl AsRef<str>]) -> Result<String, std::env::Va
 	std::env::var(secret_env_var_key(key))
 }
 
+pub async fn read_secret_opt(key: &[impl AsRef<str>]) -> Option<Result<String, std::env::VarError>> {
```

Comment:
We tend to use `Result<Option<_>>` instead of the other way around to prioritize the `?` operator